docs/mkdocs/lab.md

```diff
@@ -1,207 +1,376 @@
 
-In the case of in-cluster builds we leverage the existence of the
-[Driver Toolkit](https://github.com/openshift/driver-toolkit), a container image that includes all the necessary
-packages and tools to build a kernel module in an OpenShift cluster.
 
-Driver Toolkit image version should correspond to our Openshift server version.
-Check the [DTK documentation](https://github.com/openshift/driver-toolkit#finding-the-driver-toolkit-image-url-in-the-payload)
-for further information on this.
+## Driver Toolkit
 
-## In-cluster build from sources (multi-stage)
+We could use any image that includes the necessary libraries and dependencies for building a kernel module for our specific version. However, for ease, we can leverage Driver Toolkit (DTK), which is a container image used as a base image for building kernel modules.
 
-Create a Module called `kmm-ci-a` which will build a kernel module from sources in a remote git repository if kernel
-version in the node matches the kernel version set in the literal/regexp field.
-Once the module is built, copy it in a next step and create the definitive image which will be uploaded to the OpenShift
-internal registry and used as the Module Loader in the selected nodes.
-This multi-stage example could be useful if you need the use of extra software which is not present at DTK image but it
-is in the destination image on which the module files are copied.
+When using DTK, we should set which version of the Driver Toolkit [image](https://github.com/openshift/driver-toolkit#finding-the-driver-toolkit-image-url-in-the-payload) we are using to match the exact kernel version running on OpenShift nodes. In our case, we are using DTK with KMM and KMM will automatically determine what DTK digest should be used if the `DTK_AUTO` build argument is specified in the Dockerfile.
 
-All `Dockerfile` steps needed in order to build a module from sources should be configured in a `ConfigMap` defined
-previous to the Module object creation.
+We can get further info at DTK [repository](https://github.com/openshift/driver-toolkit#readme).
 
-<details>
-<summary>Full example</summary>
 
+## Building In-Cluster Modules from Sources
+
+In this example, we will demonstrate how to build the kernel module in our cluster using sources from a git repository. The process involves several steps:
+
+1) Adding a ConfigMap: We will create a ConfigMap to store all the necessary information for the build process, including the image we are using (DTK in this case), the remote git repository containing the sources, and the compiling commands.
+
+2) Creating the Module object: Next, we'll create the Module object itself. It will configure the registry for the resulting image, set up a secret for possible authentication, specify the previously created ConfigMap for use, and define the nodes where the pods should be scheduled.
```

Comment:
I think we should use markdown lists so:
```suggestion
1. Adding a ConfigMap: We will create a ConfigMap to store all the necessary information for the build process, including the image we are using (DTK in this case), the remote git repository containing the sources, and the compiling commands.

2. Creating the Module object: Next, we'll create the Module object itself. It will configure the registry for the resulting image, set up a secret for possible authentication, specify the previously created ConfigMap for use, and define the nodes where the pods should be scheduled.
```